### PR TITLE
Improved window and splitter geometry saving

### PR DESCRIPTION
**Pull request by ghost**
At the moment window size and splitter position is not always saved. One instance being abnormal application termination. This patch implements saving of window and splitter states when they change.

### CODE REVIEW

**Comment by dubslow** [comment]
You chose some odd locations to put new functions in widget.cpp lol. It's definitely not very well organized, but putting the two new functions between `removeGroup(Group* g, bool fake)` and `removeGroup(int groupId)` seems overly weird.

I'll fix it myself and merge.

**Comment by dubslow** [comment]
This isn't actually working for me... qTox always starts with the same geometry, no matter what. I'm just closing it in the regular manner too.

https://github.com/dubslow/qTox/commits/pr871 Note that in addition to moving the functions I had to rebase on master in order for the av to compile.

**Comment by ghost** [comment]
Thanks for rebasing \o/ As for the weird placement - header is a mess so dont blame me for cpp :p QtCreator put those functions there hehe. See in header private section:

```
void removeGroup(Group* g, bool fake = false);
void saveWindowGeometry();
void saveSplitterGeometry();
```

And thats where functions in cpp went. Anyway.

As for it not working - window geometry restoring should have been before showing window. Fixed that.

This rebasing messed things though i think :|

**Comment by dubslow** [comment]
Hmm now I see 14 commits here, not 1 or 2...

**Comment by ghost** [comment]
Yeah i merged your pr871 branch. Maybe i should not have done it?

**Comment by dubslow** [comment]
Much better, only the two commits now. Now it needs be fixed lol

**Comment by ghost** [comment]
Fixed? You mean instance where it did not work for you? Because i fixed one instance where it didnt work if "start in tray" was unchecked. Basically problem was restoring geometry after window is shown where it should happen before. Thats fixed already ;)

**Comment by dubslow** [comment]
Ah ok, my laptop is dead again so I'm basically out of commission till Sunday evening at the earliest lol

File: src/widget/widget.cpp [hunk 1]
```diff
@@ -73,7 +73,12 @@ Widget::Widget(QWidget *parent)
 void Widget::init()
 {
     ui->setupUi(this);
-    
+
+    //restore window state
+    restoreGeometry(Settings::getInstance().getWindowGeometry());
+    restoreState(Settings::getInstance().getWindowState());
+    ui->mainSplitter->restoreState(Settings::getInstance().getSplitterState());
+
```
Comment (dubslow) [inline src/widget/widget.cpp:81]: Why did you move this chunk?

https://github.com/tux3/qTox/commit/f8063fc809a93c362944deaf0cffc46f6ca17fa8

Edit: Is this the primary fix of the pr?

Comment (ghost) [inline src/widget/widget.cpp:81]: https://github.com/tux3/qTox/pull/871#issuecomment-64906517

**Comment by dubslow** [comment]
Twas rebased before merging cause core api change

Does anyone know if/how I can mark this as merged rather than closed?
